src/libraries/System.Diagnostics.DiagnosticSource/tests/W3CPropagatorTests.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class W3CPropagatorTests
+    {
+        private static DistributedContextPropagator s_w3cPropagator = DistributedContextPropagator.CreateDefaultPropagator();
+
+        [Fact]
+        public void TestW3CPropagatorBasics()
+        {
+            Assert.NotNull(s_w3cPropagator);
+            Assert.Equal(s_w3cPropagator, DistributedContextPropagator.CreateDefaultPropagator());
+            Assert.Equal(s_w3cPropagator, DistributedContextPropagator.Current);
+            Assert.Equal([ PropagatorTests.TraceParent, PropagatorTests.TraceState, PropagatorTests.Baggage ], s_w3cPropagator.Fields);
+        }
+
+        public static IEnumerable<object[]> W3CTestData()
+        {
+            // TraceState, Expected TraceState, input baggage, expected baggage string, expected baggage
```

Comment:
I'd recommend adding one test case where baggage contains a mixture of valid and invalid data to confirm that valid entries survive and only invalid entries are removed.